compiler/rustc_codegen_llvm/src/back/write.rs

```diff
@@ -974,3 +975,23 @@ fn record_artifact_size(
         self_profiler_ref.artifact_size(artifact_kind, artifact_name.to_string_lossy(), file_size);
     }
 }
+
+fn record_llvm_cgu_instructions_stats(prof: &SelfProfilerRef, llmod: &llvm::Module) {
+    if !prof.enabled() {
+        return;
+    }
+
+    let raw_stats =
+        llvm::build_string(|s| unsafe { llvm::LLVMRustModuleInstructionStats(&llmod, s) })
+            .expect("cannot get module instruction stats");
+    #[derive(serde::Deserialize)]
+    struct InstructionsStats {
+        module: String,
+        _functions: FxHashMap<String, u64>,
```

Comment:
If this field isn't used, could you please omit it until you start using it? Both here and when creating the json.